implementations/elixir/ockam/ockam/test/ockam/secure_channel/encrypted_transport_protocol/aead_aes_gcm_test.exs

```diff
@@ -88,4 +88,50 @@ defmodule Ockam.SecureChannel.EncryptedTransportProtocol.AeadAesGcmTests do
     {:ok, ciphertext, _encryptor} = Encryptor.encrypt(<<>>, plain, encryptor)
     {:ok, ^plain, _decryptor} = Decryptor.decrypt(<<>>, ciphertext, decryptor)
   end
+
+  test "out of order, exact sliding window" do
+    # We can't share the _same_ k between encryptor and decryptor on the same vault, as when the encryptor
```

Comment:
Maybe it would be clearer to describe that for this test:

 - we represent the initiator vault and the responder vault
 - we run the behavior of the `Encryptor` on the initiator node and `Decryptor` on the responder node

Reply:
would be misleading,  initiator and responder both have an encryptor and decryptor.       The encryptor on initiator side has the same key than the decryptor on responder side,    and same way around (decryptor on initiator side has the same key than encryptor on responder side).   Each flow direction  rekeys independently.